### PR TITLE
Add Phase Tickets API: `Route`, `Handler`, and `Utility Function`

### PR DESCRIPTION
**Pull request by MuhammadUmer44**
### Describe the changes:
- Implemented new API endpoint to fetch tickets by phase UUID, including route `/features/{uuid}/phase/{uuid}/tickets`, handler with authentication, and database utility function with comprehensive test coverage.

closes: #1982
closes: #1983
closes: #1984

## Issue ticket number and link:
- **Ticket Number:** [ 1982, 1983, 1984 ]
- **Link:** [ https://github.com/stakwork/sphinx-tribes-frontend/issues/1982, https://github.com/stakwork/sphinx-tribes-frontend/issues/1983, https://github.com/stakwork/sphinx-tribes-frontend/issues/1984 ]

### Evidence:

image

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome
- [x] New feature (non-breaking change which adds functionality)
- [x] I have provided a screenshot of changes in my PR

### CODE REVIEW

**Comment by MuhammadUmer44** [comment]
Hi @elraphty, @humansinstitute, @tobi-bams, 
Please review this PR. I have also added unit tests to cover all edge cases.

File: db/features.go [hunk 1]
```diff
@@ -327,3 +327,17 @@ func (db database) GetBountiesByPhaseUuid(phaseUuid string) []Bounty {
 	db.db.Model(&Bounty{}).Where("phase_uuid = ?", phaseUuid).Find(&bounties)
 	return bounties
 }
+
+func (db database) GetTicketsByPhase(featureUuid string, phaseUuid string) ([]Tickets, error) {
```
Comment (tobi-bams) [inline db/features.go:331]: @MuhammadUmer44 we are to get tickets by `phaseUuid` alone not by `phaseUuid` and `featureUuid`.

File: handlers/features.go [hunk 2]
```diff
@@ -697,3 +698,45 @@ func (oh *featureHandler) BriefSend(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	_, err := oh.db.GetFeaturePhaseByUuid(featureUuid, phaseUuid)
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		json.NewEncoder(w).Encode(map[string]string{"error": "Phase not found"})
+		return
+	}
+
+	tickets, err := oh.db.GetTicketsByPhase(featureUuid, phaseUuid)
```
Comment (tobi-bams) [inline handlers/features.go:733]: Also based on the issue related to this, you are suppose to return only the `ticketUuids` and not the entire ticket object

File: handlers/features.go [hunk 3]
```diff
@@ -697,3 +698,45 @@ func (oh *featureHandler) BriefSend(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	if _, err := uuid.Parse(phaseUuid); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(map[string]string{"error": "Invalid phase UUID format"})
+		return
+	}
+
+	_, err := oh.db.GetFeaturePhaseByUuid(featureUuid, phaseUuid)
```
Comment (tobi-bams) [inline handlers/features.go:726]: You suppose to use this new function `GetTicketsByPhase` you just created here and not this function `GetFeaturePhaseByUuid`

**Review by tobi-bams — CHANGES_REQUESTED** [review]

File: handlers/features.go (hunk 2 above)
Comment (humansinstitute) [reply handlers/features.go:733]: Hmm this should have been updated and the approach is corrrect. 

Initially I had a design to return uuids and then use a different operation to get all details. 
But it makes more sense to just return the whole object in one go whilst we're there.

File: db/features.go (hunk 1 above)
Comment (humansinstitute) [reply db/features.go:331]: There is no concept currently where we would have phases without a feature so this logic is fine.

File: handlers/features.go (hunk 2 above)
Comment (humansinstitute) [reply handlers/features.go:733]: This was a situation where the description in the ticket was correct but I'd not updated the title which still referenced UUIDs (updated)

File: db/features.go (hunk 1 above)
Comment (humansinstitute) [reply db/features.go:331]: @MuhammadUmer44 can you resolve this one so we can get it merged?

Comment (MuhammadUmer44) [reply db/features.go:331]: @humansinstitute, Sure, I am on it.

**Comment by humansinstitute** [comment]
Closing due to conflict with second pull request on same functionality.
https://github.com/stakwork/sphinx-tribes/pull/1997

**Comment by MuhammadUmer44** [comment]
> Closing due to conflict with second pull request on same functionality. #1997

@humansinstitute, I fix conflict
